skrub/_datetime_encoder.py

```diff
@@ -18,78 +22,456 @@
     "microsecond": "us",
     "nanosecond": "N",
 }
-TIME_LEVELS: list[str] = list(WORD_TO_ALIAS.keys())
-AcceptedTimeValues = Literal[
-    "year",
-    "month",
-    "day",
-    "hour",
-    "minute",
-    "second",
-    "microsecond",
-    "nanosecond",
-]
-
-
-class DatetimeEncoder(BaseEstimator, TransformerMixin):
-    """Transform each datetime column into several numeric columns \
-    for temporal features (e.g. "year", "month", "day"...).
-
-    Constant extracted features are dropped; for instance, if the year is
-    always the same in a feature, the extracted "year" column won't be added.
+TIME_LEVELS = list(WORD_TO_ALIAS)
+
+
+def _is_pandas_format_mixed_available():
+    pandas_version = pd.__version__
+    min_pandas_version = "2.0.0"
+    return parse_version(min_pandas_version) < parse_version(pandas_version)
+
+
+MIXED_FORMAT = "mixed" if _is_pandas_format_mixed_available() else None
+
+
+def to_datetime(
+    X,
+    errors="coerce",
+    **kwargs,
+):
+    """
+    Convert argument to datetime. Return the input if not datetime-parsable.
+
+    Augment :func:`pandas.to_datetime` by supporting dataframes
+    and 2d arrays inputs. It converts compatible columns to datetime, and
+    pass incompatible columns unchanged.
+
+    int, float, str, datetime, list, tuple, 1d array, and Series are defered to
+    :func:`pandas.to_datetime` directly.
+
+    Parameters
+    ----------
+    X : int, float, str, datetime, list, tuple, nd array, Series, DataFrame/dict-like
+        The object to convert to a datetime.
```

Comment:
this gives the impression the output will be a single datetime, maybe "the object to convert" and above I would start with the main use case: DataFrame or any valid argument for pd.to_datetime